### PR TITLE
VZ-11000: Refactor backup hook to work with Opster Opensearch

### PR DESCRIPTION
**Pull request by saketmht**

### CODE REVIEW

File: verrazzano-backup-hook/constants/constants.go [hunk 1]
```diff
@@ -73,14 +83,22 @@ const (
 	// IngestLabelSelector Opensearch ingest pod label selector
 	IngestLabelSelector = "app=system-es-ingest"
 
+	NewIngestStatefulSetName = "opensearch-es-ingest"
```
Comment (gaurav-05) [inline verrazzano-backup-hook/constants/constants.go:86]: Using New prefix at all places does not seem to be right, can we implement some struct or better naming

Comment (saketmht) [reply verrazzano-backup-hook/constants/constants.go:86]: I am not sure what struct would achieve here. And for naming I didn't want to use Opster so just added New prefix. Any other suggestion?

File: verrazzano-backup-hook/opensearch/interface.go [hunk 2]
```diff
@@ -58,15 +65,97 @@ type OpensearchImpl struct {
...
+	OperatorDeploymentLabelSelector string
+	// OpenSearchMasterLabel Label selector for OpenSearch master pods
+	OpenSearchMasterLabel string
+	// OpenSearchDataLabel Label selector for OpenSearch data pods
+	OpenSearchDataLabel string
+}
+
+func NewOpensearchVar(isLegacyOS bool) *OpensearchVar {
+	if isLegacyOS {
+		return &OpensearchVar{
```
Comment (gaurav-05) [inline verrazzano-backup-hook/opensearch/interface.go:130]: Does it make sense to remove declaration from constants and keep it here only to be consistent for both conditions.

Comment (saketmht) [reply verrazzano-backup-hook/opensearch/interface.go:130]: The constants are being used in tests as well. So removing them would just mean additional refactoring

Comment (gaurav-05) [reply verrazzano-backup-hook/opensearch/interface.go:130]: Ok

**Review by IshaGirdhar — DISMISSED** [review]

**Pull request merged**